docs/protocols/agent_protocol.md

```diff
@@ -0,0 +1,103 @@
+# (DRAFT) NEAR AI Agent to Agent Protocol
+
+The NEAR AI Agent Protocol is a structured format for sending messages between NEAR AI agents. Usually agents can 
+communicate between each other using natural language. For a limited number of cases, a structured format aids in 
+communication that may be fulfilled by an agent or a human.
+
+!!! warning "DRAFT"
+
+    This protocol is under active development and discussion. It will change before publication. Only 
+    develop against it if you are prepared to change your implementation.
+
+
+## Compatibility
+Agent Protocol messages are sent in Messages on Threads. This format is an extension
+of the OpenAI Thread Message format and may or may not be supported by other 
+OpenAI compatible libraries. Specifically it introduces a new message content type: `json`. 
+We recommend you use the `nearai` client to fetch these messages if you are interacting with them outside the scope of NEAR AI agents.
+
+## Communicating Capabilities
+An agent or client that supports the protocol or portions of it communicates this to agents by passing the protocol 
+in the `client_capabilities` metadata when invoking the agent. 
+
+In addition, to aid in Agent discovery, compatible agents should list the protocol in the `client_capabilities`
+field of their `metadata.json` in the agent registry.
+
+### Client Capabilities format
+Communicating full support for v1.
+```
+client_capabilities: [{
+    schema_url: "https://docs.near.ai/v1/agent_protocol.schema.json"
+}]
+```
+
+Partial support for v1 with enumerated capabilities.
+```
+client_capabilities: [{
+    schema_url: "https://docs.near.ai/v1/agent_protocol.schema.json", 
+    supported: ["request_choice", "request_signature"]
+}]
+```
+
+### List of Client Capabilities in the Agent Protocol
+- `request_choice`: The client has special handling for choice messages. 
+A UI client might display the choice in a rich UI component. An assistant agent might make a choice on behalf of the user.
+- `request_data`: The client can respond to a data request message.
+- `request_payment`: The client can respond to a payment request message with a submit payment message.
+- `request_signature`: The client can sign transactions or messages for verification.
```

Comment:
That reminds me – `request_signature()` doesn't mean a human is doing the signing, as an intermediate agent could have a key.  My first addition to this spec, once the base is drafted, would be a tool for HITL attestations where the response is signed by the client in addition to the user.